### ISSUE TITLE
joining network namespace: unknown FS magic

### ISSUE THREAD

**Issue by edsantiago** [containers/podman#16140]
Wow! This one has been hiding for months, perhaps longer. It seems to trigger often in int-test cleanup, where we do no error checking. I just noticed it in a real-live test failure:
```
$ podman [options] ps -q
time="2022-10-12T12:18:08Z" level=error msg="Joining network namespace for container <ID>: 
      retrieving network namespace at /run/user/6026/netns/netns-UUID:
      unknown FS magic on \"/run/user/6026/netns/netns-UUID\": 1021994"
Error: joining network namespace of container <ID>:
      retrieving network namespace at /run/user/6026/netns/netns-<UUID>:
      unknown FS magic on "/run/user/6026/netns/netns-<UUID>": 1021994
[AfterEach] Podman pod kill
```
Only one log to show right now, [ubuntu rootless](https://api.cirrus-ci.com/v1/artifact/task/6174187956994048/html/int-podman-ubuntu-2204-rootless-host.log.html#t--podman-pod-kill-latest-pod--1). This log file also shows three other instances of the problem, all of them non-fatal because they happen in cleanup. Search in-page for "fs magic".

**Comment by Luap99** [comment]
Honestly I do not understand why we use the NetNs interface on the Container State object. This requires us to open the netns to use it. If you look at the users of this interface they all call .Path() so they only want the path which means the the open is completely unnecessary and adds overhead. We should just use the path, this would also remove the need for the code
[replaceNetNS()](https://github.com/containers/podman/blob/main/libpod/boltdb_state_linux.go#L15) function which creates the error you are seeing.
@mheon WDYT?

**Comment by mheon** [comment]
We're only storing the path in the database right now, so this conversion should not result in a breakage of DB format. We'll still blow up horribly when trying to do anything with the old namespace (e.g. start a container), but that will be a lot less often than it happening any time we try and pull a namespace out of the database. Sounds like a reasonable approach to me.

**Comment by edsantiago** [comment]
This is a thorny one, because it isn't actually triggering a test failure: the error happens in `Cleanup()`, which for complicated reasons (which I don't actually remember any more, just that they were complicated and unsolvable) cannot do exit-status checks. https://github.com/containers/podman/blob/6428ff180a3757f399b4e0fef0222c324d780eed/test/e2e/common_test.go#L537-L539

So, the only way I have to log these is if _another_ test causes a triple-fail ginkgo failure that causes a CI respin. Hence, really hard for me to monitor. Here's a list (obviously incomplete) from the last two weeks:

* Oct 20, [fedora 36 rootless](https://api.cirrus-ci.com/v1/artifact/task/5412605295591424/html/int-podman-fedora-36-rootless-host.log.html#t--podman-pod-clone-shm-size--1)
* Oct 21, [ubuntu 2204 rootless](https://api.cirrus-ci.com/v1/artifact/task/5397531035697152/html/int-podman-ubuntu-2204-rootless-host.log.html#t--podman-pod-clone-shm-size--1)
* Oct 23, [fedora 36 rootless](https://api.cirrus-ci.com/v1/artifact/task/5737254827065344/html/int-podman-fedora-36-rootless-host.log.html#t--podman-pod-clone-shm-size--1)

All failures are in the `podman pod clone --shm-size` e2e test, which I find interesting.

**Comment by edsantiago** [comment]
Another one November 9, [f36 rootless](https://api.cirrus-ci.com/v1/artifact/task/6141700557504512/html/int-podman-fedora-36-rootless-host.log.html#t--podman-pod-clone-shm-size--1).

As mentioned before, this one is not trackable using my regular flake logging.

**Comment by Luap99** [comment]
I can work on it this week. I know that there are a lot of users seeing this error as well.

**Comment by smokeyhallow** [comment]
This just happened to me! Is this resolved? I am on ` podman 3.4.4` via WSL2 in Windows 10
